src/tools/miri/tests/native-lib/pass/ptr_write_access.rs

```diff
@@ -5,118 +5,192 @@
+    test_swap_nested_ptr();
+
+    test_swap_tuple();
+
+    test_overwrite_dangling();
 
-    test_inner_alloc_exposed();
+    test_expose_triple();
 }
 
-fn test_modify_int() {
+// Test function that modifies an int.
+fn test_increment_int() {
     extern "C" {
-        fn modify_int(ptr: *mut i32);
+        fn increment_int(ptr: *mut i32);
     }
 
     let mut x = 11;
-    unsafe { modify_int(&mut x) };
 
+    unsafe { increment_int(&mut x) };
     assert_eq!(x, 12);
 }
 
+// Test function that initializes an int.
 fn test_init_int() {
     extern "C" {
-        fn init_int(ptr: *mut i32);
+        fn init_int(ptr: *mut i32, val: i32);
     }
 
     let mut x = MaybeUninit::<i32>::uninit();
+    let val = 21;
+
     let x = unsafe {
-        init_int(x.as_mut_ptr());
+        init_int(x.as_mut_ptr(), val);
         x.assume_init()
     };
-
-    assert_eq!(x, 21);
+    assert_eq!(x, val);
 }
 
+// Test function that initializes an array.
 fn test_init_array() {
     extern "C" {
-        fn init_array(ptr: *mut i32, len: usize);
+        fn init_array(ptr: *mut i32, len: usize, val: i32);
     }
 
     const LEN: usize = 3;
-
     let mut array = MaybeUninit::<[i32; LEN]>::uninit();
+    let val = 31;
+    
     let array = unsafe {
-        init_array(array.as_mut_ptr().cast::<i32>(), LEN);
+        init_array(array.as_mut_ptr().cast::<i32>(), LEN, val);
         array.assume_init()
     };
-
-    assert_eq!(array, [31; LEN]);
+    assert_eq!(array, [val; LEN]);
 }
 
-fn test_init_interior_mutable() {
+// Test function that initializes an int pointed to by an immutable static.
+fn test_init_static_inner() {
+    #[repr(C)]
+    struct SyncPtr {
+        ptr: *mut i32
+    }
+    unsafe impl Sync for SyncPtr {}
+
     extern "C" {
-        fn init_interior_mutable(pptr: *const UnsafeInterior);
+        fn init_static_inner(s_ptr: *const SyncPtr, val: i32);
     }
 
-    #[repr(C)]
-    struct UnsafeInterior {
-        mut_ptr: *mut i32
+    static mut INNER: MaybeUninit<i32> = MaybeUninit::uninit();
+    #[allow(static_mut_refs)]
+    static STATIC: SyncPtr = SyncPtr { ptr: unsafe { INNER.as_mut_ptr() } };
+    let val = 41;
+
+    let inner = unsafe {
+        init_static_inner(&STATIC, val);
+        INNER.assume_init()
+    };
+    assert_eq!(inner, val);
+}
+
+// Test function that writes a pointer and exposes the alloc of its int argument.
+fn test_expose_int() {
+    extern "C" {
+        fn expose_int(int_ptr: *const i32, pptr: *mut *const i32);
     }
-    unsafe impl Sync for UnsafeInterior {}
-    
-    let mut x = MaybeUninit::<i32>::uninit();
-    let unsafe_interior = UnsafeInterior { mut_ptr: x.as_mut_ptr() };
-    unsafe { init_interior_mutable(&unsafe_interior) };
 
-    assert_eq!(unsafe { x.assume_init() }, 51);
+    let x = 51;
+    let mut ptr = std::ptr::null();
+
+    unsafe { expose_int(&x, &mut ptr) };
+    assert_eq!(unsafe { *ptr }, x);
 }
 
+// Test function that swaps two pointers and exposes the alloc of an int.
```

Comment:
```suggestion
/// Test function that swaps two pointers and exposes the alloc of an int.
```
Same for all the other comments explaining a function.